### PR TITLE
[#1988] Anonymous first-item: bind the fill to registration, fork the landing CTA

### PR DESCRIPTION
**Pull request by denisvmedia**
## Problem

The anonymous "add your first item before you have an account" flow (follow-up to #1988 / #2005) let a logged-out visitor pick a **currency** inside the item form — but an anonymous visitor can't know which **group** (and therefore which currency) they'll land in after auth. The post-login resolver then silently dropped the drafted item into *whatever* group they signed into, reinterpreting prices against that group's currency. Fine for a brand-new user (the item *is* their first inventory), but confusing/wrong for someone who already has an inventory.

## Approach

Reframe the anonymous fill as **new-user onboarding**, and tell returning users to just log in:

- **Landing forks up front.** Two clear paths: **"Add your first item"** (fill → hand off to registration) and **"Log in"** (returning users go straight to their existing inventory — no anonymous draft). A low-emphasis footer link offers a no-item **"Create an account"** path.
- **Hand-off targets `/register`, not `/login`.** Registration requires email verification (no auto-login), so the new user still completes via login — and the `localStorage` pending-first-item marker survives the whole `register → verify → sign-in` round-trip, so the existing replay still fires at login.
- **Currency comes from the item (unchanged).** When the freshly-registered user has zero groups, `FirstItemResolver` already auto-creates a "Main" group seeded with the item's own currency, then replays the draft into it. No ambiguity.

## Why (almost) no backend changes

The currency-aware group auto-creation already exists: `FirstItemResolver` (0-groups branch) → `useCreateGroup({ group_currency })` → `GroupService.CreateGroup`. This PR is frontend-only.

## Changes

- `LandingPage.tsx` — two-card fork ("Add your first item" / "Log in") + footer "Create an account" link; copy/icons reworked.
- `AnonymousCommodityDialog.tsx` — `handleSubmit` / `handleSaveDraft` redirect to `/register?redirect=/welcome`.
- `landing.json` — dropped the now-misleading "no account needed to start"; `cards.browse`/`loginCta` → `cards.login`/`registerCta`.
- `LoginPage.tsx` — see the follow-up fix below.
- Comments in `firstItemHandoff.ts` updated.

## Follow-up: fix post-login redirect race (surfaced by CI e2e)

The first e2e run failed deterministically in all 3 browsers — the replay landed on `/g/<group>` instead of the item, because of a race the original #1988 flow had masked. The hand-off now reaches `/login` **without** a `?redirect` (via `/register` → verify-email → `/login`, which drops the query). `finalizeLogin` still routes to `/welcome` off the pending marker, but `LoginPage`'s `isAuthenticated` guard *also* fires on the post-login re-render and — with no redirect param — navigated to `/`, unmounting `FirstItemResolver` mid-replay.

Fix: gate that guard on `!loginMutation.isSuccess`, so a just-completed form login defers to `finalizeLogin` as the **sole** post-login navigator (a pre-existing/deep-linked session still redirects exactly as before). This also hardens the real new-user path, which legitimately reaches `/login` without the redirect param. Added a `LoginPage` regression test.

## Edge case (deliberate)

An existing user who *fills then logs in* (e.g. clicks "sign in" from the register page) still has the item replayed into their existing group. That's safe — commodities carry their own `original_price_currency` and the resolver re-runs price conversion against the real group — and it's a rare, user-initiated path rather than the default.

## Tests

- vitest: `LandingPage` (new cards + register-link nav), `AnonymousCommodityDialog` (register hand-off), and a new `LoginPage` regression test (fresh login + pending marker + no `?redirect` → `/welcome`) — green. `typecheck` / `eslint` / `prettier --check` clean.
- e2e `anonymous-first-item.spec.ts`: landing → fill → `/register` → `/login` → replay into the group, verifying no-data-loss + cross-user isolation.

## Summary by CodeRabbit

* **New Features**
  * Landing page now prominently offers account creation as the primary action for new users.

* **Bug Fixes**
  * Fixed redirect flow for users starting item creation without an account—now properly routes to registration.
  * Fixed item drafts created before sign-in to properly resume after login, preventing loss of user input.

### CODE REVIEW

File: e2e/tests/anonymous-first-item.spec.ts [hunk 1]
```diff
@@ -2,8 +2,12 @@
  * Anonymous "add your first item before login" journey (#1988).
  *
  * Exercises the no-data-loss guarantee end-to-end: a logged-out visitor
- * fills the create form on the public landing page, is sent to log in, and
- * after auth the drafted item is replayed into their group with nothing lost.
+ * fills the create form on the public landing page, is sent to REGISTER (the
+ * anonymous fill is framed as new-user onboarding), and after auth the
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply e2e/tests/anonymous-first-item.spec.ts:6]: Good catch — fixed in 4338e99f. Rewrote the header comment to match current behavior: the landing "Add your first item" card is **always** shown; `public_scan` only gates the in-dialog AI photo-scan *step* (the public scan endpoint is mounted only when the operator opts in). This spec drives that step and dismisses it via "Fill manually", so it stubs `GET /feature-flags` to report `public_scan: true` purely so the step is present to dismiss — the scan itself never runs.
